internal/provisioner/kops_provisioner_installation_sli.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package provisioner
+
+import (
+	"context"
+	"time"
+
+	"github.com/mattermost/mattermost-cloud/k8s"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	slothv1 "github.com/slok/sloth/pkg/kubernetes/api/sloth/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (provisioner *KopsProvisioner) createInstallationSLI(clusterInstallation *model.ClusterInstallation, k8sClient *k8s.KubeClient) error {
+	wait := 60
+	sli := &slothv1.PrometheusServiceLevel{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterInstallation.InstallationID,
+			Labels: map[string]string{
+				"app":     "kube-prometheus-stack",
+				"release": "prometheus-operator",
+			},
+		},
+		Spec: slothv1.PrometheusServiceLevelSpec{
+			Service: clusterInstallation.InstallationID,
+			Labels: map[string]string{
+				"owner": "sreteam",
+			},
+			SLOs: []slothv1.SLO{
+				{
+					Name:        "requests-availability",
+					Objective:   99.9,
+					Description: "Availability metric for mattermost API",
+					SLI: slothv1.SLI{Events: &slothv1.SLIEvents{
+						ErrorQuery: "sum(rate(mattermost_api_time_count{job='mm-" + clusterInstallation.Namespace[0:4] + "',code=~'(5..|429)'}[{{.window}}]))",
+						TotalQuery: "sum(rate(mattermost_api_time_count{job='mm-" + clusterInstallation.Namespace[0:4] + "'}[{{.window}}]))",
```

Comment:
There is a helper method for making the CI name `makeClusterInstallationName()` instead of the manual process done here and in the lines below.

It's probably best to use it in case we change how resource names are made in the future.